ostd/src/sync/rcu/monitor.rs

```diff
@@ -1,23 +1,22 @@
 // SPDX-License-Identifier: MPL-2.0
 
 use alloc::collections::VecDeque;
-use core::sync::atomic::{
-    AtomicBool,
-    Ordering::{self, Relaxed},
-};
+use core::sync::atomic::{AtomicBool, Ordering};
 
 use crate::{
-    cpu,
-    cpu::{AtomicCpuSet, CpuSet},
+    cpu::{AtomicCpuSet, CpuSet, PinCurrentCpu},
     prelude::*,
     sync::SpinLock,
+    task::disable_preempt,
 };
 
 /// A RCU monitor ensures the completion of _grace periods_ by keeping track
 /// of each CPU's passing _quiescent states_.
 pub struct RcuMonitor {
-    is_monitoring: AtomicBool,
-    state: SpinLock<State>,
+    cur_queue_dropping: AtomicBool,
+    cpus_passed_quiescent: AtomicCpuSet,
+    cur_waitqueue: SpinLock<Callbacks>,
+    next_waitqueue: SpinLock<Callbacks>,
```

Comment:
Since we have a widely-used type whose name is also `WaitQueue`, I think these two fields should be renamed to avoid confusion.